examples/webview2/src/main.rs

```diff
@@ -0,0 +1,622 @@
+        let js: Vec<u16> = js.encode_utf16().collect();
+        let js = HString::from_wide(&js);
+
+        let webview = self.webview.as_ref();
+        let (tx, rx) = mpsc::channel();
+        webview
+            .ExecuteScriptAsync(js)?
+            .SetCompleted(AsyncOperationCompletedHandler::new(move |op, _status| {
+                if let Some(op) = op {
+                    tx.send(op.GetResults()?).expect("send over mpsc channel");
+                }
+                Ok(())
+            }))?;
+
+        wait_with_pump(rx)?;
+        Ok(self)
+    }
+
+    pub fn dispatch<F>(&self, f: F) -> Result<&Self>
+    where
+        F: FnOnce(WebView) + Send + 'static,
+    {
+        self.tx.send(Box::new(f)).expect("send the fn");
+
+        unsafe {
+            WindowsAndMessaging::PostThreadMessageA(
+                self.thread_id,
+                WindowsAndMessaging::WM_APP,
+                WPARAM(0),
+                LPARAM(0),
+            );
+        }
+        Ok(self)
+    }
+
+    pub fn bind<F>(&self, name: &str, f: F) -> Result<&Self>
+    where
+        F: FnMut(u64, Vec<Value>) + 'static,
+    {
+        self.bindings
+            .lock()
+            .unwrap()
+            .insert(String::from(name), Box::new(f));
+
+        let js = String::from(
+            r#"
+            (function() {
+                var name = '"#,
+        ) + name
+            + r#"';
+                var RPC = window._rpc = (window._rpc || {nextSeq: 1});
+                window[name] = function() {
+                    var seq = RPC.nextSeq++;
+                    var promise = new Promise(function(resolve, reject) {
+                        RPC[seq] = {
+                            resolve: resolve,
+                            reject: reject,
+                        };
+                    });
+                    window.external.invoke({
+                        id: seq,
+                        method: name,
+                        params: Array.prototype.slice.call(arguments),
+                    });
+                    return promise;
+                }
+            })()"#;
+
+        self.init(&js)
+    }
+
+    pub fn resolve(&self, id: u64, status: i32, result: Value) -> Result<&Self> {
+        let result = result.to_string();
+
+        self.dispatch(move |webview| {
+            let method = match status {
+                0 => "resolve",
+                _ => "reject",
+            };
+            let js = format!(
+                r#"
+                window._rpc[{}].{}({});
+                window._rpc[{}] = undefined;"#,
+                id, method, result, id
+            );
+
+            webview.eval(&js).expect("eval return script");
+        })
+    }
+
+    fn set_window_webview(hwnd: HWND, webview: Box<WebView>) {
+        unsafe {
+            WindowsAndMessaging::SetWindowLongPtrA(
+                hwnd,
+                WINDOW_LONG_PTR_INDEX::GWLP_USERDATA,
+                Box::into_raw(webview) as _,
```

Comment:
`webview` will be dropped at the end of the function. Is that what we want or does that cause a dangling pointer?

Reply:
I thought `Box::into_raw` consumed the `Box` and avoided calling `drop`, is that incorrect?

Reply:
Double checked docs, I think this is safe.